ledger-core/instrumentation/insapp/server.go

```diff
@@ -47,6 +57,14 @@ func (s *Server) readConfig(cfgPath string) configuration.Configuration {
 	return *cfgHolder.Configuration
 }
 
+func (s *Server) SetKeyStoreFactory(factoryFn KeyStoreFactory) {
+	s.keyStoreFactory = factoryFn
+}
+
+func (s *Server) SetCertManagerFactory(factoryFn CertManagerFactory) {
+	s.certManagerFactory = factoryFn
+}
+
```

Comment:
There should be no setters for individual parts. 
Necessary factories should be provided at construction first. Then, if you need to override'em please use "impose" approach (see AppCompartment.SetImposer for reference)

Reply:
remove setters